sql/expression/function/aggregation/avg_test.go

```diff
@@ -132,3 +132,79 @@ func TestAvg_NULL(t *testing.T) {
 	require.NoError(err)
 	require.Equal(nil, eval(t, avgNode, buffer))
 }
+
+func TestAvg_NUMS_AND_NULLS(t *testing.T) {
+	require := require.New(t)
+	ctx := sql.NewEmptyContext()
+
+	avgNode := NewAvg(expression.NewGetField(0, sql.Uint64, "col1", true))
+	buffer := avgNode.NewBuffer()
+	require.Zero(avgNode.Eval(ctx, buffer))
+
+	err := avgNode.Update(ctx, buffer, sql.NewRow(2))
+	require.NoError(err)
+
+	err = avgNode.Update(ctx, buffer, sql.NewRow(nil))
+	require.NoError(err)
+
+	err = avgNode.Update(ctx, buffer, sql.NewRow(4))
+	require.NoError(err)
+
+	require.Equal(3.0, eval(t, avgNode, buffer))
+}
+
+func TestAvg_Distinct(t *testing.T) {
+	require := require.New(t)
+	ctx := sql.NewEmptyContext()
+
+	ad := expression.NewDistinctExpression(expression.NewGetField(0, nil, "myfield", false))
+	avg := NewAvg(ad)
+
+	// first validate that the expression's name is correct
+	require.Equal("AVG(DISTINCT myfield)", avg.String())
+
+	testCases := []struct {
+		name     string
+		rows     []sql.Row
+		expected interface{}
+	}{
+		{
+			"string int values",
+			[]sql.Row{{"1"}, {"1"}, {"2"}, {"2"}, {"3"}, {"3"}, {"4"}, {"4"}},
+			float64(2.5),
+		},
+		{
+			"string float values",
+			[]sql.Row{{"2.0"}, {"2.0"}, {"3.0"}, {"4.0"}, {"4.0"}},
+			float64(3.0),
+		},
+		{
+			"float values",
+			[]sql.Row{{2.0}, {2.0}, {3.}, {4.}},
+			float64(3.0),
+		},
+		{
+			"no rows",
+			[]sql.Row{},
+			nil,
+		},
+		{
```

Comment:
Need a test case with some null and some non null values (esp. since you changed the null handling code)